### ISSUE TITLE
 kubectl log/exec function refactored and enhanced

### ISSUE THREAD

**Issue by ZhengXinwei-F** [kubeedge/kubeedge#4996]
**What would you like to be added/modified**:

 kubectl log/exec function refactored and enhanced

**Why is this needed**:

The present version of the kubectl logs/exec function has a certificate issuance issue and requires iptables operation on the kube-apiserver node. These are insufficiently reasonable.

**Comment by sdghchj** [comment]
The issue of getting metrics through the metrics-server:
If cloudcore insists nodeName as the tunnel session ID of an edge node, there must be a way the http requests from a metrics-server tell cloudcore.
The value of header `Host` in a http request from metrics-server, I guess, may depend on the settings `--kubelet-preferred-address-types` of metrics-server and api-server.

I have tried to set the `Host` header with nodeName in metrics-server, but they refused to merge [metrics-server-pr-1069](https://github.com/kubernetes-sigs/metrics-server/pull/1069).

Or I wonder, if we set --kubelet-preferred-address-types="Hostname,xxx" with Hostname at first position for api-server and metrics-server, would this be fixed?

**Comment by ZhengXinwei-F** [comment]
> The issue of getting metrics through the metrics-server: If cloudcore insists nodeName as the tunnel session ID of an edge node, there must be a way the http requests from a metrics-server tell cloudcore. The value of header `Host` in a http request from metrics-server, I guess, may depend on the settings `--kubelet-preferred-address-types` of metrics-server and api-server.
> 
> I have tried to set the `Host` header with nodeName in metrics-server, but they refused to merge [metrics-server-pr-1069](https://github.com/kubernetes-sigs/metrics-server/pull/1069).
> 
> Or I wonder, if we set --kubelet-preferred-address-types="Hostname,xxx" with Hostname at first position for api-server and metrics-server, would this be fixed?

Yes, if we set --kubelet-preferred-address-types="Hostname,xxx" before the apiserver connects to the kubelet, the issue of edge node IP duplication can be resolved.
However, it may cause an issue when cloudcore passes a request to the cloud node, cloudcore must be able to parse the node hostname and forward the request to the proper node. 
Also, when we deploy KubeEdge on a public cloud, it is difficult to modify the kube-apiserver settings in the master node.
